### DISCUSSION TITLE
FR: Add `javascript.exported` config, as a counterpart to `javascript.globals`

### DISCUSSION THREAD

**Discussion by chrisgrieser** [biomejs/biome#1147] [Ideas]
biome currently has an allowlist for `noUndeclaredVariables` in the form of `javascript.globals`, where we can globally define exceptions for `noUndeclaredVariables` instead of disabling the rule at every occurrence of the variable.

I'd like to propose introducing `javascript.exported` which should act as an allowlist for `noUnusedVariables` in the very same way. Other tools like for example [the lua LSP allow for such an allowlist](https://luals.github.io/wiki/settings/#diagnosticsunusedlocalexclude). eslint also supports such an allowlist, [but in the form of the rule option for `no-unused-vars`](https://eslint.org/docs/latest/rules/no-unused-vars#varsignorepattern).

The use case of such a setting / rule option would be to allow for variables that are implicitly exported, for example the `run` function in JXA. (Right now, the alternative is to literally add a disabling-comment at every single occurrence of such `run` functions.)

**Comment by Conaclos** [comment]
Could you add an example? or link to an example?

Why not explicitly export the variable?

**Reply by chrisgrieser** [reply]
> Why not explicitly export the variable?

JXA does not support any sort of export/import statements (it basically always runs in single-file-mode). `run` is a special variable, if running a JXA file, the `run` function is automatically called, even if you have no `run()` statement in your code.

> Could you add an example? or link to an example?

Given a file  `foobar.js` with the following content, `osascript l- JavaScript "foobar.js"` will return `hello world`:
```js
function run() {
  console.log("hello  world")
}
```

**Reply by ematipico** [reply]
I'm not really sure what JXA is, but `globals` wasn't **only** meant for rules, but for other tools such as transformer and/or minifier.

If we want to add an exception, it should be inside the rule
